interactions.go

```diff
@@ -7,8 +7,230 @@ import (
+	ApplicationCommandOptionSubCommandGroup
+	ApplicationCommandOptionString
+	ApplicationCommandOptionInteger
+	ApplicationCommandOptionBoolean
+	ApplicationCommandOptionUser
+	ApplicationCommandOptionChannel
+	ApplicationCommandOptionRole
+)
+
+// ApplicationCommandOption is representing an option/subcommand/subcommands group.
+type ApplicationCommandOption struct {
+	Type        ApplicationCommandOptionType `json:"type"`
+	Name        string                       `json:"name"`
+	Description string                       `json:"description,omitempty"`
+	// Default     bool                              `json:"default"`
+	Required bool                              `json:"required"`
+	Choices  []*ApplicationCommandOptionChoice `json:"choices"`
+	Options  []*ApplicationCommandOption       `json:"options"`
+}
+
+// ApplicationCommandOptionChoice is representing slash-command's option choice.
+type ApplicationCommandOptionChoice struct {
+	Name  string      `json:"name"`
+	Value interface{} `json:"value"`
+}
+
+// InteractionType is representing interaction type.
+type InteractionType uint8
+
+// Interaction types
+const (
+	InteractionPing = InteractionType(iota + 1)
+	InteractionApplicationCommand
+)
+
+// Interaction is representing interaction with application.
+type Interaction struct {
+	ID        string                            `json:"id"`
+	Type      InteractionType                   `json:"type"`
+	Data      ApplicationCommandInteractionData `json:"data"`
+	GuildID   string                            `json:"guild_id"`
+	ChannelID string                            `json:"channel_id"`
+	Member    *Member                           `json:"member"`
+	Token     string                            `json:"token"`
+	Version   int                               `json:"version"`
+}
+
+// ApplicationCommandInteractionData is representing interaction data for application command.
+type ApplicationCommandInteractionData struct {
+	ID      string                                     `json:"id"`
+	Name    string                                     `json:"name"`
+	Options []*ApplicationCommandInteractionDataOption `json:"options"`
+}
+
+// ApplicationCommandInteractionDataOption is representing an option of application's command.
+type ApplicationCommandInteractionDataOption struct {
+	Name string `json:"name"`
+	// Contains the value specified by InteractionType
+	Value   interface{}                                `json:"value,omitempty"`
+	Options []*ApplicationCommandInteractionDataOption `json:"options,omitempty"`
+}
+
+// IntValue is utility function for casting option value to integer
+func (o ApplicationCommandInteractionDataOption) IntValue() (i int64) {
+	if v, ok := o.Value.(float64); ok {
+		i = int64(v)
+	}
+	return
+}
+
+// UintValue is utility function for casting option value to unsigned integer
+func (o ApplicationCommandInteractionDataOption) UintValue() (i uint64) {
+	if v, ok := o.Value.(float64); ok {
```

Comment:
Same as my previous comment, but this should be `o.Value.(uint64)`.

Reply:
No.